pkg/relayer/proxy/proxy.go

```diff
@@ -59,60 +69,79 @@ func NewRelayerProxy(
 	clientCtx sdkclient.Context,
 	keyName string,
 	keyring keyring.Keyring,
-
+	nativeServicesListenAddress map[serviceId]string,
 	// TODO_INCOMPLETE(@red-0ne): Uncomment once the BlockClient interface is available.
 	// blockClient blocktypes.BlockClient,
 ) RelayerProxy {
 	accountQuerier := accounttypes.NewQueryClient(clientCtx)
 	supplierQuerier := suppliertypes.NewQueryClient(clientCtx)
 	sessionQuerier := sessiontypes.NewQueryClient(clientCtx)
-	providedServices := buildProvidedServices(ctx, supplierQuerier)
 	servedRelays, servedRelaysProducer := channel.NewObservable[*types.Relay]()
 
 	return &relayerProxy{
 		// TODO_INCOMPLETE(@red-0ne): Uncomment once the BlockClient interface is available.
-		// blockClient:          blockClient,
-		keyName:              keyName,
-		keyring:              keyring,
-		accountsQuerier:      accountQuerier,
-		supplierQuerier:      supplierQuerier,
-		sessionQuerier:       sessionQuerier,
-		providedServices:     providedServices,
-		servedRelays:         servedRelays,
-		servedRelaysProducer: servedRelaysProducer,
+		// blockClient:       blockClient,
+		keyName:                     keyName,
+		keyring:                     keyring,
+		accountsQuerier:             accountQuerier,
+		supplierQuerier:             supplierQuerier,
+		sessionQuerier:              sessionQuerier,
+		nativeServicesListenAddress: nativeServicesListenAddress,
+		servedRelays:                servedRelays,
+		servedRelaysProducer:        servedRelaysProducer,
 	}
 }
 
-// Start starts all supported proxies and returns an error if any of them fail to start.
+// Start concurrently starts all advertised relay servers and returns an error if any of them fails to start.
 func (rp *relayerProxy) Start(ctx context.Context) error {
-	panic("TODO: implement relayerProxy.Start")
+	// The provided services map is built from the supplier's on-chain advertised information,
+	// which is a runtime parameter that can be changed by the supplier.
+	// Build the provided services map at Start instead of NewRelayerProxy to avoid having to
+	// return an error from the constructor.
+	err := rp.BuildProvidedServices(ctx)
+	if err != nil {
+		return err
+	}
+
+	startGroup, gctx := errgroup.WithContext(ctx)
+
+	for _, relayServer := range rp.advertisedRelayServers {
+		for _, svr := range relayServer {
+			server := svr // create a new variable scoped to the anonymous function
```

Comment:
Does this not work given that we reinitiliaze `svr` on every iteration of the forloop?

Reply:
If we use `svr` directly we have a `loop variable svr captured by func literal` warning
![image](https://github.com/pokt-network/poktroll/assets/231488/8a4f2754-878b-49b3-90e3-75f06c5270a1)